ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vlan_interfaces.py

```diff
@@ -60,11 +61,30 @@ def _get_vlan_interface_config_for_svi(self, svi, vrf, tenant) -> dict:
             # Only set VARP if ip_address is set
             vlan_interface_config["ip_virtual_router_addresses"] = svi.get("ip_virtual_router_addresses")
 
+        vlan_virtual_address = svi.get("ip_address_virtual")
         if vlan_interface_config.get("ip_virtual_router_addresses") is None:
             # Only set Anycast GW if VARP is not set
-            vlan_interface_config["ip_address_virtual"] = svi.get("ip_address_virtual")
+            vlan_interface_config["ip_address_virtual"] = vlan_virtual_address
             vlan_interface_config["ip_address_virtual_secondaries"] = svi.get("ip_address_virtual_secondaries")
 
+        pim_config_ipv4 = {}
+        if default(get(svi, "l3_multicast.enabled"), get(vrf, "_l3_multicast_enabled")):
+            if self._mlag:
+                pim_config_ipv4["sparse_mode"] = True
+            else:
+                vlan_interface_config["ip_igmp"] = True
+
+            if vlan_virtual_address:
```

Comment:
All this logic should only be for the case of MLAG right?

Reply:
Thanks, have fixed this.